org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -299,31 +344,48 @@ public static File getGradleJavaHomeFile() {
 		return null;
 	}
 
-	protected void startSynchronization(Path projectFolder, IProgressMonitor monitor) {
+	protected IStatus startSynchronization(Path projectFolder, IProgressMonitor monitor) {
 		File location = projectFolder.toFile();
 		boolean shouldSynchronize = shouldSynchronize(location);
 		if (shouldSynchronize) {
 			BuildConfiguration build = getBuildConfiguration(projectFolder);
-			SynchronizationResult result = GradleCore.getWorkspace().createBuild(build).synchronize(monitor);
-			if (!result.getStatus().isOK()) {
-				JavaLanguageServerPlugin.log(result.getStatus());
+			GradleBuild gradleBuild = GradleCore.getWorkspace().createBuild(build);
+			SynchronizationResult result = gradleBuild.synchronize(monitor);
+			IStatus resultStatus = result.getStatus();
+			if (isFailedStatus(resultStatus)) {
+				try {
+					BuildEnvironment environment = gradleBuild.withConnection(connection -> connection.getModel(BuildEnvironment.class), monitor);
+					GradleEnvironment gradleEnvironment = environment.getGradle();
+					String gradleVersion = gradleEnvironment.getGradleVersion();
+					File javaHome = getJavaHome(getPreferences());
+					String javaVersion;
+					if (javaHome == null) {
+						javaVersion = System.getProperty("java.version");
+					} else {
+						StandardVMType type = new StandardVMType();
+						javaVersion = type.readReleaseVersion(javaHome);
+					}
+					if (GradleCompatibilityChecker.isIncompatible(GradleVersion.version(gradleVersion), javaVersion)) {
+						URI uri = projectFolder.toUri();
+						Path path = Paths.get(uri);
```

Comment:
projectFolder is already a `Path` instance, no need to convert it to another `Path` again.

Reply:
since we pass projectPath between client and server, https://github.com/eclipse/eclipse.jdt.ls/pull/1965/commits/87298b495b093411006826cc69f1775704fea69d removes the uri instance and will use one projectFolder to avoid misunderstanding.